### ISSUE TITLE
Windows VM stoping response

### ISSUE THREAD

**Issue by aerowork** [QubesOS/qubes-issues#2456]
#### Qubes OS version (e.g., `R3.1`):

3.2

#### Affected TemplateVMs (e.g., `fedora-23`, if applicable):

windows 7 HVM template
with QWT 3.2.2-3
seamlees mode

---

### Actual behavior:

After updating to 3.2.2-3 QWT from current-testing repo VM behavior comes less stable. Update according description in google mail-list. All works fine about 2-3 hours, than app windows stop to respond any mouse or keyboard action. I have to reboot VM but it freeze on this action even if CPU load comes to 0%.

In dom0 logs I see "Failed to connect to gui-agent" (giud.win.log) and 
```
libvchan_is_oef
vchan closed
reconnecting
vchan closed
```

In windows VM qubesdb-daemon:
```
[20161124.123557.077-152-E] QpsReaderThread: ReadFile failed with error 109: The pipe has been ended.
[20161124.123557.077-152-W] QpsReaderThread: [6] read failed
[20161124.123557.077-1392-W] QpsWriterThread: [6] client is disconnecting, exiting
[20161124.123557.077-1636-W] QpsRead: [6] client is disconnected
[20161124.123557.077-1636-E] pipe_thread_client: QpsRead failed with error 0: The operation completed successfully.
[20161124.123557.077-1636-W] pipe_thread_client: read from client 6 failed
[20161124.123557.295-1700-E] XcGnttabUnmapForeignPages: IOCTL_XENIFACE_GNTTAB_UNMAP_FOREIGN_PAGES failed
[20161124.123557.295-1700-E] XcGnttabUnmapForeignPages: Error: 0x490
[20161124.123557.295-1700-E] XcGnttabUnmapForeignPages: IOCTL_XENIFACE_GNTTAB_UNMAP_FOREIGN_PAGES failed
[20161124.123557.295-1700-E] XcGnttabUnmapForeignPages: Error: 0x490
[20161124.123557.295-1700-E] ErrUEF: Unhandled exception c0000005 at 000007FEF941220F, flags: 0
[20161124.123557.295-1700-E] ErrUEF: Parameter 0: 0000000000000001
[20161124.123557.295-1700-E] ErrUEF: Parameter 1: 0000000000300014
```

### General notes:

---

**Comment by aerowork** [comment]
Looks like this in not QWT problem, restored from backup VM with QWT  3.2.1-2 and have same issue.
I think that was starting after dom0 update:

```
qubes-core-dom0 - 3.2.11.1->3.2.12.1
qubes-db - 3.2.1.1 -> 3.2.3.1
```

**Comment by aerowork** [comment]
I could find out that bug occurs after exactly 30 min without actions in VM. I reproduce it on fresh windows (win7x64) after installing QWT.

Downgrade qubes-core-dom0 and qubes-db, qubes-db-libs does not solve the problem.

Will be grateful to get any help, its really interferes to work. Thanx.

**Comment by aerowork** [comment]
Fixed by enabling "Power" windows service. 

Need to correct doc
https://www.qubes-os.org/doc/windows-template-customization/

Instead of disabling service need to disable any idle action in power scheme.

**Comment by andrewdavidwong** [comment]
How would that affect the [Disable hibernation](https://www.qubes-os.org/doc/windows-template-customization/#disable-hibernation) section? Should that section be removed?

**Comment by 3n7r0p1** [comment]
> Fixed by enabling "Power" windows service.

Using QWT 3.2.2 on **R3.1** with `power disabled`: I do not experience this issue over 12hr uptimes.
Your issue may be 3.2-specific. 

> I reproduce it on fresh windows (win7x64) after installing QWT.

Is this with Windows updates installed? (anecdotal) I find that QWT stability varies wildly with the Windows update status.

**Comment by aerowork** [comment]
> How would that affect the Disable hibernation section? Should that section be removed?

I guess disable hibernation should to do, because sleep and hibernation actions while idle anyway need to disable.

> Your issue may be 3.2-specific.
Guess so. 

> Is this with Windows updates installed? (anecdotal) I find that QWT stability varies wildly with the Windows update status.

Win7 sp1 + kb2534111, kb976902 from distrib.

**Comment by andrewdavidwong** [comment]
> I guess disable hibernation should to do, because sleep and hibernation actions while idle anyway need to disable.

Sorry, I didn't quite follow that. Mind rephrasing?

**Comment by aerowork** [comment]
I mean in addition to disable hibernation should set 0 (never) all timeouts in power scheme config

**Comment by andrewdavidwong** [comment]
> I mean in addition to disable hibernation should set 0 (never) all timeouts in power plan config

Well, I'm asking specifically about updating the documentation, since this is now a documentation issue. The documentation currently states that the "Power" service should be disabled. (It states this both in the "Windows services" section and in the "Disable hibernation" section.) You're saying that, contrary what the documentation currently states, the "Power" service should be enabled. It also sounds like you're confirming that hibernation should, in fact, be disabled (as the docs currently state).

However, this leaves it unclear what the "Disable hibernation" section should now say. The first step in that section states: "Ensure that you disabled the Power service...". So, if we're now saying to *enable* the Power service, the page will be self-contradictory.

In other words, you're saying that we need to correct the doc, but it's not clear exactly how it should be corrected in a consistent way.

**Comment by aerowork** [comment]
Ok, 
in truth I don`t understand why current doc section "Disable Hibernation" requires to ensure that service is disabled (and stopped) before set configuration by powercfg. To my mind that is not necessary and simple execution "powercfg -h off" is OK. Anyway this condition couldn't be implemented, so I propose to remove this from doc.

In addition, while "Power" service stays enabled need to configure Power Options from defaults. Possible should rename "Disable Hibernation" section to "Power Options", fix disabling conditions and add recommendation to configure Advanced Power option and disable:
- Sleep timeout;
- Display off timeout;
- Hard disk turn off timeout;
